registry/remote/example_test.go

```diff
@@ -0,0 +1,624 @@
+	}
+
+	digest := "sha256:00e5ffa7d914b4e6aa3f1a324f37df0625ccc400be333deea5ecaa199f9eff5b"
+	// resolve the blob descriptor to obtain the size of the blob
+	descriptor, err := repo.Resolve(ctx, digest)
+	if err != nil {
+		panic(err)
+	}
+	rc, err := repo.Fetch(ctx, descriptor)
+	if err != nil {
+		panic(err)
+	}
+	defer rc.Close() // don't forget to close
+	pulled, err := io.ReadAll(rc)
+	if err != nil {
+		panic(err)
+	}
+	// verify the fetched content
+	if descriptor.Digest != ocidigest.FromBytes(pulled) || descriptor.Size != int64(len(pulled)) {
+		panic(err)
+	}
+
+	fmt.Println(string(pulled))
+	// Output:
+	// Example manifest content
+}
+
+// ExampleRepository_FetchReference_manifestByTag gives example snippets for downloading a manifest by tag with only one API call.
+func ExampleRepository_FetchReference_manifestByTag() {
+	reg, err := remote.NewRegistry(host)
+	if err != nil {
+		panic(err)
+	}
+	ctx := context.Background()
+	repo, err := reg.Repository(ctx, exampleRepositoryName)
+	if err != nil {
+		panic(err)
+	}
+
+	tag := "latest"
+	// specify the expected manifest media type for the blob
+	reg.ManifestMediaTypes = append(reg.ManifestMediaTypes, ocispec.MediaTypeImageManifest)
+	descriptor, rc, err := repo.FetchReference(ctx, tag)
+	if err != nil {
+		panic(err)
+	}
+	defer rc.Close() // don't forget to close
+	pulledBlob, err := io.ReadAll(rc)
+	if err != nil {
+		panic(err)
+	}
+	// verify the fetched content
+	if descriptor.Digest != ocidigest.FromBytes(pulledBlob) || descriptor.Size != int64(len(pulledBlob)) {
+		panic(err)
+	}
+
+	fmt.Println(string(pulledBlob))
+
+	// Output:
+	// Example manifest content
+}
+
+// ExampleRepository_FetchReference_manifestByDigest gives example snippets for downloading a manifest by digest.
+func ExampleRepository_FetchReference_manifestByDigest() {
+	reg, err := remote.NewRegistry(host)
+	if err != nil {
+		panic(err)
+	}
+	ctx := context.Background()
+	repo, err := reg.Repository(ctx, exampleRepositoryName)
+	if err != nil {
+		panic(err)
+	}
+
+	digest := "sha256:00e5ffa7d914b4e6aa3f1a324f37df0625ccc400be333deea5ecaa199f9eff5b"
+	descriptor, rc, err := repo.FetchReference(ctx, digest)
+	if err != nil {
+		panic(err)
+	}
+	defer rc.Close() // don't forget to close
+	pulled, err := io.ReadAll(rc)
+	if err != nil {
+		panic(err)
+	}
+	// verify the fetched content
+	if descriptor.Digest != ocidigest.FromBytes(pulled) || descriptor.Size != int64(len(pulled)) {
+		panic(err)
+	}
+
+	fmt.Println(string(pulled))
+
+	// Output:
+	// Example manifest content
+}
+
+// ExampleRepository_Fetch_layer gives example snippets for downloading a layer blob by digest.
+func ExampleRepository_Fetch_layer() {
+	reg, err := remote.NewRegistry(host)
+	if err != nil {
+		panic(err)
+	}
+	ctx := context.Background()
+	repo, err := reg.Repository(ctx, exampleRepositoryName)
+	if err != nil {
+		panic(err)
+	}
+
+	descriptor, err := repo.Blobs().Resolve(ctx, digest.FromBytes([]byte(exampleLayer)).String())
```

Comment:
`digest.FromBytes([]byte(exampleLayer)).String()` is confusing. Can we replace it with a pre-computed variable or constant?